### PR TITLE
Remove experimental/streaming globals, add streaming example

### PR DESCRIPTION
**Pull request by jmacd**
@rghetia this is a cleanup in `experimental/streaming` which is a first step towards the work on global initialization in RFC 0005 and the work on globals in https://github.com/jmacd/opentelemetry-go/pull/3.

The updates in `README.md` should be paired with your updates in `example/http`. Then, I'd like `README.md` to eventually model after the Python README:

https://github.com/open-telemetry/opentelemetry-python/blob/master/README.md

### CODE REVIEW

**Comment by rghetia** [comment]
LGTM

File: experimental/streaming/exporter/observer/observer.go [hunk 1]
```diff
@@ -85,73 +82,42 @@ const (
...
-var (
-	observerMu sync.Mutex
-	observers  atomic.Value
-
-	sequenceNum uint64
-)
-
-func NextEventID() EventID {
-	return EventID(atomic.AddUint64(&sequenceNum, 1))
+type Exporter struct {
```
Comment (krnowak) [inline experimental/streaming/exporter/observer/observer.go]: Should that type go to the `experimental/streaming/exporter` package then?

**Comment by krnowak** [comment]
And one more - please add `/experimental/streaming/example/basic/basic` to `.gitignore`.

File: experimental/streaming/exporter/observer/observer.go (hunk 1 above)
Comment (jmacd) [reply experimental/streaming/exporter/observer/observer.go]: Done.

**Comment by rghetia** [comment]
rebase to include #139 and then run make circle-ci again to update the go.mod/go.sum files.

**Pull request merged**